src/Microsoft.Health.Fhir.Core/Features/Persistence/ResourceDeserializer.cs

```diff
@@ -38,6 +40,95 @@ public ResourceElement Deserialize(ResourceWrapper resourceWrapper)
             return resource;
         }
 
+        public ResourceElement Deserialize(RawResourceElement rawResourceElement)
+        {
+            EnsureArg.IsNotNull(rawResourceElement, nameof(rawResourceElement));
+
+            ResourceElement resource = DeserializeRaw(rawResourceElement);
+
+            return resource;
+        }
+
+        /// <summary>
+        /// Get the raw data from resourceWrapper as a string and JsonDocument.
+        /// If the RawResource's VersionSet or LastUpdatedSet are false, then the RawResource's data will be updated
+        /// to have them set to the values in the ResourceWrapper
+        /// </summary>
+        /// <param name="resourceWrapper">Input ResourceWrapper to convert to a JsonDocument</param>
+        /// <returns>A <see cref="JsonDocument"/></returns>
+        public static (string, JsonDocument) DeserializeToJsonDocument(ResourceWrapper resourceWrapper)
+        {
+            EnsureArg.IsNotNull(resourceWrapper, nameof(resourceWrapper));
+
+            if (resourceWrapper.RawResource.LastUpdatedSet && resourceWrapper.RawResource.VersionSet)
+            {
+                return (resourceWrapper.RawResource.Data, JsonDocument.Parse(resourceWrapper.RawResource.Data));
+            }
+
+            var jsonDocument = JsonDocument.Parse(resourceWrapper.RawResource.Data);
+
+            using (var ms = new MemoryStream())
```

Comment:
We should use a pool of these (RecycleableMemoryStream)

Reply:
Good call, I didn't know about this.

Reply:
This got moved to ResourceWrapperExtensions, but should be resolved.